ngraph_bridge/ngraph_encapsulate_op.cc

```diff
@@ -454,159 +458,51 @@ void NGraphEncapsulateOp::ComputeUsingParallelExecutor(OpKernelContext* ctx) {
-                               tensor_manager->GetNumberOfInputs(),
-                               " and num of "
-                               "input tensors from ctxt ",
-                               tf_input_tensors.size(), " do not match"));
+                               num_of_inputs, " and Ctx->num_inputs() ",
+                               ctx->num_inputs(), " do not match"));
+  OP_REQUIRES(
+      ctx, num_of_inputs == tf_input_tensors.size(),
+      errors::Internal("Num of inputs from TensorManager ", num_of_inputs,
+                       " and num of "
+                       "input tensors from ctxt ",
+                       tf_input_tensors.size(), " do not match"));
+  OP_REQUIRES(
+      ctx, num_of_inputs == ng_exec->get_parameters().size(),
+      errors::Internal("Num of inputs from TensorManager ", num_of_inputs,
+                       " and num of "
+                       "parameters from exec ",
+                       ng_exec->get_parameters().size(), " do not match"));
 
-  OP_REQUIRES(ctx, tensor_manager->GetNumberOfOutputs() == ctx->num_outputs(),
+  OP_REQUIRES(ctx, num_of_outputs == ctx->num_outputs(),
               errors::Internal("Num of outputs from TensorManager ",
-                               tensor_manager->GetNumberOfOutputs(),
-                               " and Ctx->num_outputs()", ctx->num_outputs(),
-                               " do not match"));
-  OP_REQUIRES(ctx, tensor_manager->GetNumberOfOutputs() ==
-                       ng_exec->get_results().size(),
+                               num_of_outputs, " and Ctx->num_outputs()",
+                               ctx->num_outputs(), " do not match"));
+  OP_REQUIRES(ctx, num_of_outputs == ng_exec->get_results().size(),
               errors::Internal("Num of outputs from TensorManager ",
-                               tensor_manager->GetNumberOfOutputs(),
-                               "and number of exec outputs ",
+                               num_of_outputs, "and number of exec outputs ",
                                ng_exec->get_results().size(), " do not match"));
 
-  // create inputs, outputs, pipelineId
-  int num_of_inputs = tensor_manager->GetNumberOfInputs();
-  int num_of_outputs = tensor_manager->GetNumberOfInputs();
-  int current_iter_pipeline_depth = get<0>(io_tensors);
+  // Get pipelined input output tensors for this iteration
+  std::tuple<int, PipelinedTensorVector, PipelinedTensorVector>
+      pipelined_io_tensors;
+  OP_REQUIRES_OK(ctx, GetPipelinedIOTensorsReadyForExecution(
+                          ctx, tf_input_tensors, pipelined_tensor_store,
+                          tensor_manager, pipelined_io_tensors));
+
+  int current_iter_pipeline_depth = get<0>(pipelined_io_tensors);
   vector<shared_ptr<ng::runtime::Tensor>> ng_inputs(num_of_inputs);
   vector<shared_ptr<ng::runtime::Tensor>> ng_outputs(num_of_outputs);
 
   // Assume All inputs and outputs are pipelined
```

Comment:
Get rid of this comment

Reply:
Second thoughts, this comment is still valid
All inputs and outputs are still pipelined.
Of all these pipelined inputs some are prefetched and that has been taken care of in this PR

Before this PR,
All inputs and outputs were pipelined. And all inputs were assumed to be also prefetched.

Reply:
In that case Get rid of word 'Assume',  and mention what you just wrote in the comment.